gadget/params.c

```diff
@@ -489,10 +510,31 @@ void read_parameter_file(char *fname)
 
         /*Lyman-alpha forest parameters*/
         All.QuickLymanAlphaProbability = param_get_double(ps, "QuickLymanAlphaProbability");
-        All.HeliumHeatOn = param_get_int(ps, "HeliumHeatOn");
-        All.HeliumHeatThresh = param_get_double(ps, "HeliumHeatThresh");
-        All.HeliumHeatAmp = param_get_double(ps, "HeliumHeatAmp");
-        All.HeliumHeatExp = param_get_double(ps, "HeliumHeatExp");
+
+        param_get_string2(ps, "TreeCoolFile", All.TreeCoolFile);
+        param_get_string2(ps, "UVFluctuationfile", All.UVFluctuationFile);
+        param_get_string2(ps, "MetalCoolFile", All.MetalCoolFile);
+
+        /*Cooling rate network parameters*/
+        locals.coolpar.CMBTemperature = All.CP.CMBTemperature;
```

Comment:
What about moving these into `set_cooling_params(ps)`?

Reply:
I think I was trying to make a clean separation between "parameters needed for initialisation" and "parameters needed for calculation". 

I think I want the parameters for calculation to be stored per-module, and the parameters for initialisation to continue to be stored in All (or something like it). At the moment one of the things that I don't like about All is that it combines "things for init" with "parameters for calculation" and even worse "parameters that change every timestep". It will take a while, but I'd ultimately like the three things to be separated (ideally the every-timestep parameters are local variables of course).

Reply:
Would it make sense to have two levels then? first divide by module, then within a module, by frequency? Because the module decides which parameter is used more frequently.